src/crimson/osd/object_context.h

```diff
@@ -107,7 +107,30 @@ class ObjectContext : public ceph::common::intrusive_lru_base<
     });
   }
 
+  boost::intrusive::list_member_hook<> list_hook;
+  uint64_t list_link_cnt = 0;
+
 public:
+
+  template <typename ListType>
+  void link_me(ListType&& list) {
+    if (!list_link_cnt)
+      list.push_back(*this);
```

Comment:
nit, could you add curly brackets around the statement. like 
```c++
if (!list_link_cnt) {
  list.push_back(*this);
}
```
? for better readability.